### PR TITLE
Fix Infinity values in timestamp columns

### LINKED ISSUE #86: Value of infinity in timestamp columns breaks > v.1.5.0
**Problem**
I have just tried to upgrade from v.1.5.0 to the latest available version to use fresh version of knex. However, [181e410b9dd60abb8d6fffe4f5d9c54765c39a8a](https://github.com/markusahlstrand/knex-data-api-client/commit/181e410b9dd60abb8d6fffe4f5d9c54765c39a8a#diff-e9726d469bba0bec8608c095ad5c361aeb95ea53afadd7ab44b462a2eb5e7467R11) seems to break the possible value of `infinity` in PostgreSQL database, which is valid value according to the [spec](https://www.postgresql.org/docs/9.0/datatype-datetime.html).

**Background**
"**8.5.1.4. Special Values**
PostgreSQL supports several special date/time input values for convenience, as shown in [Table 8-13](https://www.postgresql.org/docs/9.0/datatype-datetime.html#DATATYPE-DATETIME-SPECIAL-TABLE). The values infinity and -infinity are specially represented inside the system and will be displayed unchanged; but the others are simply notational shorthands that will be converted to ordinary date/time values when read. (In particular, now and related strings are converted to a specific time value as soon as they are read.) All of these values need to be enclosed in single quotes when used as constants in SQL commands."

**What is causing it**
Now, the timestamp value of `"+292278994-08-16 23:00:00"` (representing the value of infinity) is passed to the match method in https://github.com/markusahlstrand/knex-data-api-client/blob/master/src/types.js#L15 and the array destructor fails with error of "object null is not iterable (cannot read property Symbol(Symbol.iterator)".

I am not sure who is responsible for converting infinity value to `"+292278994-08-16 23:00:00"`: AWS, data-api-client or this project. Should this library take into account the possibility of inifinity values? With version 1.5.0 the infinity values are working as expected.

### PR DESCRIPTION
**Pull request by jaska120**
Fixes #86 

Due to regression described in the linked issue, this library could not fetch timestamp values of infinity from PostgreSQL timestamp columns. This PR fixes that and adds more capabilities over Infinity values:

1. Tests are now performed with the same version as expected in the peerDependencies.
2. Tests for inserting and selecting Infinity values have been added. Knex allows with pg library to insert both `Infinity` and `"infinity"` values to timestamp columns with the same result, but this library could only insert `"infinity"` string values.
3. Correct inserts & select operations are added in this PR. Please double check them, because I did not have any easy way to deploy this version to AWS and therefore the changes are made a little bit white box way. Especially: 1) are the bindings correct way to allow numeric Infinity values? 2) Is the date string comparison method feasible in your opinion?

I am happy to revise, if needed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "eslint-config-prettier": "8.3.0",
     "eslint-plugin-import": "2.24.2",
     "eslint-plugin-prettier": "4.0.0",
-    "knex": "0.21.1",
+    "knex": "^0.95.6",
```
Comment (markusahlstrand) [inline package.json:53]: We kept this at 0.21 earlier as Objection.js hadn't updated yet, but they have also switched to a newer version now so this should be fine.

**Review by markusahlstrand — APPROVED** [review]

File: src/types.js [hunk 2]
```diff
@@ -8,13 +8,25 @@ function applyRecord(columnMetadata, record) {
...
-            /^(\d{1,4})-(\d{1,2})-(\d{1,2}) (\d{1,2}):(\d{1,2}):(\d{1,2})(?:.(\d{1,3}))?(?:\d{1,3})?$/,
+            /^([\+|\-]?\d+)-(\d{1,2})-(\d{1,2}) (\d{1,2}):(\d{1,2}):(\d{1,2})(?:.(\d{1,3}))?(?:\d{1,3})?$/,
           );
 
+          // Data Api may return timestamps out of range of JS max timestamp value of +-8640000000000000
+          // Year comparison is performed first and if the year is in range, then ISO date strings may be compared
+          const yearNumber = Number(year);
+          const isoDateString = `${year}-${month}-${day}T${hour}:${minute}:${second}.${millisecond.padEnd(3, 0)}Z`;
+          if (yearNumber > 275760 || isoDateString >= '+275760-09-13T00:00:00.000Z') {
+            return Infinity;
```
Comment (markusahlstrand) [inline src/types.js:24]: I guess this should be assigned to the property rather than returned? I'll have a look

**Review by markusahlstrand — CHANGES_REQUESTED** [review]
We need to fix the lint errors before merging. I'll have a quick look

**Comment by markusahlstrand** [comment]
I created a new branch based on this PR to make all tests pass.

File: src/types.js (hunk 2 above)
Comment (jaska120) [reply src/types.js:24]: Sorry for my mistake and I see you have corrected this in another PR.

**Pull request merged**